src/responses/Response.cpp

```
@@ -1,67 +1,98 @@
+    }
+    else if (req.getMethod() == "DELETE") {
         handleDeleteRequest(req);
-    } else {
-        setStatus(405, "Method Not Allowed");
-        setBody("Only GET, POST, and DELETE methods are supported");
+    }
+    else {
+        setStatus(405);
+        setBody("405 - Method Not Allowed");
     }
 }
 
 void Response::handleGetRequest(const Request& req, const ServerConfig& config) {
     std::string filename = config.root + req.getUri();
-    if (filename == config.root + "/")
-		filename = config.root + "/index.html";
-
-	std::cout << "Filename: " << filename << std::endl;
-
+    if (filename == config.root + "/") {
+        filename = config.root + "/index.html";
+    }
     std::string content = readFile(filename);
     if (content.empty()) {
-        setStatus(404, "Not Found");
+        setStatus(404);
         setBody("404 - Page not found");
     } else {
-        setStatus(200, "OK");
+        setStatus(200);
         setBody(content);
         addHeader("Content-Type", getMimeType(filename));
     }
 }
 
-
 void Response::handlePostRequest(const Request& req) {
-        setStatus(200, "OK");
-        setBody("POST request received for URI: " + req.getUri() + "\nBody: " + req.getBody());
-        addHeader("Content-Type", "text/plain");
+    setStatus(200);
+    setBody("POST request received for URI: " + req.getUri() + "\nBody: " + req.getBody());
+    addHeader("Content-Type", "text/plain");
 }
 
 void Response::handleDeleteRequest(const Request& req) {
-    setStatus(200, "OK");
+    setStatus(200);
     setBody("DELETE request received for URI: " + req.getUri());
     addHeader("Content-Type", "text/plain");
 }
 
-void Response::setStatus(int code, const std::string& message) {
+void Response::setStatus(int code) {
     _statusCode = code;
-    _statusMessage = message;
+    std::map<int, std::string>::iterator it = _httpErrors.find(code);
+    _statusMessage = (it != _httpErrors.end()) ? it->second : "Unknown";
```

Would rather throw an error here since not finding a code should never happen (this would indicate an error in the code we wrote)